components/salsa-macro-rules/src/setup_input_struct.rs

```diff
@@ -89,6 +92,12 @@ macro_rules! setup_input_struct {
 
                 type Revisions = [$zalsa::Revision; $N];
                 type Durabilities = [$zalsa::Durability; $N];
+
+                $(
+                    fn heap_size(value: &Self::Fields, _panic_if_missing: $zalsa::PanicIfHeapSizeMissing) -> usize {
```

Comment:
Instead of threading the `PanicIfHeapSizeMissing` everywhere, you could add a `has_heap_size` method to `Ingredient` and do a single check that all ingredients return `true` if `PanicIfHeapSizeMissing::Yes` is set. You'd need a `HAS_HEAP_SIZE` constant on all the configurations, but that should be simple to add.

Reply:
I actually like the current way more. It is simpler (especially if we want to show the name of the problematic ingredient), and the threading is not that bad. Do you have a preference?

Reply:
I don't think @ibraheemdev meant to expose that, just as an internal implementation detail.

If we do expose that, how do we? Do we change `IngredientInfo::size_of_fields` to `Option<usize>`?

Reply:
Why is the memo aggregation a problem here? If we split `size_of_fields` I think that should be sufficient to detect whether or not the heap size was tracked for a given memo type.